eng/pipelines/templates/jobs/archetype-sdk-tests-generate.yml

```diff
@@ -0,0 +1,62 @@
+parameters:
+- name: AdditionalParameters
+  type: object
+- name: CloudConfig
+  type: object
+  default: {}
+- name: MatrixConfigs
+  type: object
+  default: []
+
+jobs:
+- job: generate_matrix
+  variables:
+    displayNameFilter: $[ coalesce(variables.jobMatrixFilter, '.*') ]
+  pool:
+    name: azsdk-pool-mms-ubuntu-1804-general
+    vmImage: MMSUbuntu18.04
+  displayName: Generate Job Matrix
+  steps:
+    - ${{ each config in parameters.MatrixConfigs }}:
+      - ${{ if eq(config.GenerateVMJobs, 'true') }}:
```

Comment:
Should we try to generalize this to be a matrix name/grouping of some sort instead of being hard-coded to be VM or container?

Reply:
In this case, the only reason for this switching is because of the issue where as long as the container key exists then devops will try to run on a container (even with an empty value). I'm not super happy with the naming/switching but am open to better or more general solutions.

Reply:
Since it is only 2 cases currently lets keep it harded-coded but if we find ourselves adding more we should think about generalizing.

Reply:
Yeah I think when we start wanting azure functions that will be a good time to do it.